aiida/tools/archive/implementations/sqlite_zip/writer.py

```diff
@@ -164,7 +165,13 @@ def _stream_binary(
         if compression is not None:
             kwargs['compression'] = zipfile.ZIP_DEFLATED if compression else zipfile.ZIP_STORED
             kwargs['level'] = compression
-        with self._zip_path.joinpath(name).open('wb', **kwargs) as zip_handle:
+
+        # compute the file size of the handle
+        handle.seek(0, os.SEEK_END)
+        file_size = handle.tell()
+        handle.seek(0)
```

Comment:
I guess in most cases the position of the handle will be `0` here, but it is not guaranteed right? To be absolutely safe, you might want to do the following:
```suggestion
        position = handle.tell()
        handle.seek(0, os.SEEK_END)
        file_size = handle.tell()
        handle.seek(position)
```